test/new-e2e/containers/eks_test.go

```diff
@@ -0,0 +1,499 @@
+			"kube_namespace:workload-dogstatsd",
+		},
+		[]*regexp.Regexp{
+			regexp.MustCompile(`^kube_deployment:dogstatsd-udp$`),
+			regexp.MustCompile(`^kube_namespace:workload-dogstatsd$`),
+			regexp.MustCompile(`^kube_ownerref_kind:replicaset$`),
+			regexp.MustCompile(`^kube_ownerref_name:dogstatsd-udp-[[:alnum:]]+$`),
+			regexp.MustCompile(`^kube_qos:Burstable$`),
+			regexp.MustCompile(`^kube_replica_set:dogstatsd-udp-[[:alnum:]]+$`),
+			regexp.MustCompile(`^pod_name:dogstatsd-udp-[[:alnum:]]+-[[:alnum:]]+$`),
+			regexp.MustCompile(`^pod_phase:running$`),
+			regexp.MustCompile(`^series:`),
+		},
+	)
+}
+
+func (suite *eksSuite) TestPrometheus() {
+	// suite.T().Parallel()
+
+	// Test Prometheus check
+	suite.testMetric("prom_gauge",
+		[]string{
+			"kube_deployment:prometheus",
+			"kube_namespace:workload-prometheus",
+		},
+		[]*regexp.Regexp{
+			regexp.MustCompile(`^container_id:`),
+			regexp.MustCompile(`^container_name:prometheus$`),
+			regexp.MustCompile(`^display_container_name:prometheus`),
+			regexp.MustCompile(`^endpoint:http://.*:8080/metrics$`),
+			regexp.MustCompile(`^git.commit.sha:`),                                                       // org.opencontainers.image.revision docker image label
+			regexp.MustCompile(`^git.repository_url:https://github.com/DataDog/test-infra-definitions$`), // org.opencontainers.image.source   docker image label
+			regexp.MustCompile(`^image_id:ghcr.io/datadog/apps-prometheus@sha256:`),
+			regexp.MustCompile(`^image_name:ghcr.io/datadog/apps-prometheus$`),
+			regexp.MustCompile(`^image_tag:main$`),
+			regexp.MustCompile(`^kube_container_name:prometheus$`),
+			regexp.MustCompile(`^kube_deployment:prometheus$`),
+			regexp.MustCompile(`^kube_namespace:workload-prometheus$`),
+			regexp.MustCompile(`^kube_ownerref_kind:replicaset$`),
+			regexp.MustCompile(`^kube_ownerref_name:prometheus-[[:alnum:]]+$`),
+			regexp.MustCompile(`^kube_qos:Burstable$`),
+			regexp.MustCompile(`^kube_replica_set:prometheus-[[:alnum:]]+$`),
+			regexp.MustCompile(`^pod_name:prometheus-[[:alnum:]]+-[[:alnum:]]+$`),
+			regexp.MustCompile(`^pod_phase:running$`),
+			regexp.MustCompile(`^series:`),
+			regexp.MustCompile(`^short_image:apps-prometheus$`),
+		},
+	)
+}
+
+func (suite *eksSuite) testMetric(metricName string, filterTags []string, expectedTags []*regexp.Regexp) {
+	suite.Run(fmt.Sprintf("%s{%s}", metricName, strings.Join(filterTags, ",")), func() {
+		// suite.T().Parallel()
+
+		suite.EventuallyWithTf(func(collect *assert.CollectT) {
+			metrics, err := suite.Fakeintake.FilterMetrics(
+				metricName,
+				fakeintake.WithTags[*aggregator.MetricSeries](filterTags),
+			)
+			if err != nil {
+				collect.Errorf("%w", err)
+				return
+			}
+			if len(metrics) == 0 {
+				collect.Errorf("No `%s{%s}` metrics yet", metricName, strings.Join(filterTags, ","))
+				return
+			}
+
+			// Check tags
+			if err := assertTags(metrics[len(metrics)-1].GetTags(), expectedTags); err != nil {
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
Do you think it makes sense to have this exact tags assertion on the fakeintake client side ?

Reply:
I discussed a bit about this subject with @ogaca-dd and he also thought that such a feature would make sense inside the fakeintake client API.

I’m curious to know if the other new-e2e users have the same need.
As @DataDog/container-integrations, we maintain the tagger and this component has a lot of tag sources, so, the various asserted tags are in fact testing different pieces of code.

Reply:
👋 

This is the initial PR https://github.com/DataDog/integrations-core/pull/4035 from a while ago, it has some example and context.

The latest version of the code is here: https://github.com/DataDog/integrations-core/blob/master/datadog_checks_base/datadog_checks/base/stubs/similar.py